### ISSUE TITLE
when posting a list of string remove [] to the key, i.e how to remove [] for the key when posting a list of data using FormData.fromMap()

### ISSUE THREAD

**Issue by sreevardhanreddi** [flutterchina/dio#633]
### New Issue Checklist

- [X] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

```
data = {'labels':[1,2,3,4]}
FormData formData = FormData.fromMap(data);

// the data is posted as 
{'labels[]':[1,2,3,4]}
// but I want it to be posted as 
{'labels':[1,2,3,4]}
```

**Comment by Chunlee17** [comment]
I have the same issue with dio version: 3.0.8 and 2.1.16

**Comment by sreevardhanreddi** [comment]
> I have the same issue with dio version: 3.0.8 and 2.1.16

I kind of found a workaround, hope it helps others, but I would like the library to provide an option to configure this.

to fix this when you are posting a list of data

```
     /// this condition is for fields with List of data
    /// [Dio] appends [] to the keys which have a value of List
    /// which works for some backends,
    /// but Django and some other backends do not accept it
   /// data['lables'] = [1,2,3,4]
    
    if (data['label'] != null || data['label'].length > 0) {
      final labels = data['label'];
      for (var item in labels) {
        formData.fields.add(MapEntry('label', item));
      }
    }
```
now I am recieving it as {'labels':[1,2,3,4]} in the backend.

**Comment by romatroskin** [comment]
Why not using the name directly, so those back-ends who needs to append brackets can use the data = {'labels[]':[1,2,3,4]}, specifying name directly, and all other wouldn't be broken.

**Comment by michaldev** [comment]
The same problem.

**Comment by neokree** [comment]
Someone is working on this request? I'm not able to use a third party API because of this. 

It would be very useful if I could set an option to disable this behaviour

**Comment by Ruslanbek0809** [comment]
I have searched for answer for a while. This worked for me:
Step 1. Create a model for it using [https://pub.dev/packages/json_serializable](json_serializable) package by making your model `explicitToJson` to true as it explained here [https://docs.flutter.dev/development/data-and-backend/json#consuming-json_serializable-models](official flutter docs) . 
Step 2. When you want to post it don't use **formData** instead use **data** property without **formData**  like this `data: jsonEncode(yourModel));`. In real api it should look like this: `await dio.post(api/order/', data: jsonEncode(yourModel));`

**Comment by abdalazeez1** [comment]
when i encode list like this 
 data['json_key'] = json.encode(listOfInt);///or something  else String .....
now can send list 
MapEntry(json_key: [1])

**Comment by patana93** [comment]
In my case using json_serializable lib. I have resolved like this:
```

final jsonFile = MyModel.toJson();
    jsonFile['my_list_obj_field_on_db'] = json.encode(jsonFile["my_list_obj_field_on_db"]);
   
 final data = FormData.fromMap(jsonFile);

    final Response<Map<String, dynamic>> response;
    try {
      response = await _dio.post(path, data: data);
    } catch (err) {
    ... //other code
    
```
